lib/folio_sync/archives_space/manual_updater.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+# This class is meant to be used in a one-time manual step
+module FolioSync
+  module ArchivesSpace
+    class ManualUpdater
+      PAGE_SIZE = 200
+
+      def initialize(instance_key)
+        @logger = Logger.new($stdout)
+        @aspace_client = FolioSync::ArchivesSpace::Client.new(instance_key)
+        @folio_client = FolioSync::Folio::Client.instance
+        @instance_key = instance_key
+        @csv_file_path = "tmp/#{instance_key}_updated_aspace_resources_#{Time.zone.now.strftime('%Y%m%d%H%M%S')}.csv"
+        initialize_csv_file
+      end
+
+      def retrieve_and_sync_aspace_resources
+        @aspace_client.fetch_all_repositories.each do |repo|
+          next log_repository_skip(repo) unless repo['publish']
+
+          repo_id = extract_id(repo['uri'])
+          fetch_from_repo_and_update_resources(repo_id)
+        end
+      end
+
+      def determine_potential_hrid(resource)
+        if @instance_key == 'cul'
+          resource['id_0']
+        elsif @instance_key == 'barnard' && resource['user_defined']['string_1']
+          resource['user_defined']['string_1']
+        end
+      end
+
+      def fetch_from_repo_and_update_resources(repo_id)
+        query_params = build_query_params
+
+        @aspace_client.retrieve_resources_for_repository(repo_id, query_params) do |resources|
+          resources.each do |resource|
+            next if resource['suppressed']
+
+            potential_hrid = determine_potential_hrid(resource)
+            source_record = @folio_client.find_source_record(instance_record_hrid: potential_hrid)
+
+            if source_record
+              update_aspace_record(resource, repo_id)
+              write_to_csv(resource, potential_hrid)
```

Comment:
It might be better to open the CSV file handle before the `@aspace_client.retrieve_resources_for_repository(repo_id, query_params) do |resources|` line, and close it after the corresponding `end`.  The current logic appears to open and close the CSV file once for each resource that's updated.  Maybe it's not that big of a deal since this is for a one-time manual process, but under normal circumstances we wouldn't want to repeatedly open and close a CSV file once per line.

Reply:
For a one-time operation I'm fine with keeping things as they are, but if this code might be used in the future or kept in the repo long term it would be better to update.  My comment above is more of a good practice recommendation for future scenarios.

Reply:
Good point. I updated this logic to open the CSV, write to it, and close it in one operation.